### PR TITLE
Split HeroBonus.h to pieces, part2

### PR DESCRIPTION
**Pull request by rilian-la-te**
This is split of enums suggested by @IvanSavenko 

Includes #2105, should be reviewed after merge of that PR.

### CODE REVIEW

**Comment by rilian-la-te** [comment]
@IvanSavenko will undraft this after merging of #2105

**Review by Nordsoft91 — APPROVED** [review]
Consider to place everything related to bonuses into dedicated namespace

**Comment by Nordsoft91** [comment]
I'm talking about all Bonus-related classes, not just enums

**Comment by Nordsoft91** [comment]
Benefit is identifiers isolation of big sub-system.
it's not necessity, feel free to ignore. Was only comment to consider such option.

File: lib/bonuses/BonusEnum.h [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+	BONUS_VALUE(BASE_NUMBER)\
+	BONUS_VALUE(PERCENT_TO_ALL)\
+	BONUS_VALUE(PERCENT_TO_BASE)\
+	BONUS_VALUE(PERCENT_TO_SOURCE) /*Adds value only to bonuses with same source*/\
+	BONUS_VALUE(PERCENT_TO_TARGET_TYPE) /*Adds value only to bonuses with SourceType target*/\
+	BONUS_VALUE(INDEPENDENT_MAX) /*used for SPELL bonus */\
+	BONUS_VALUE(INDEPENDENT_MIN) //used for SECONDARY_SKILL_PREMY bonus
+
+
+enum class BonusTypeID
```
Comment (IvanSavenko) [inline lib/bonuses/BonusEnum.h:218]: Perhaps simply BonusType?
All other enumerations use same form - BonusXXX, without 'ID' suffix

Comment (rilian-la-te) [reply lib/bonuses/BonusEnum.h:218]: @IvanSavenko BonusTypeID was already used in CBonusTypeHandler with `using BonusTypeID = Bonus::Type`. Should I change past uses of BonusTypeID to BonusType too?

Comment (IvanSavenko) [reply lib/bonuses/BonusEnum.h:218]: I'd say yes.
Only two occurences from what I see.

Comment (rilian-la-te) [reply lib/bonuses/BonusEnum.h]: Okay.

**Review by IvanSavenko — APPROVED** [review]

**Pull request merged**